src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -94,6 +97,16 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
         require(msg.sender == parentVault, NotAuthorized());
         require(marketParams.loanToken == asset, LoanAssetMismatch());
 
+        if (shares[marketId] == 0 && assets > 0) {
+            for (uint256 i = 0; i < allMarketParams.length; i++) {
+                if (Id.unwrap(allMarketParams[i].id()) == Id.unwrap(marketId)) {
+                    allMarketParams[i] = allMarketParams[allMarketParams.length - 1];
+                    allMarketParams.pop();
+                    break;
+                }
+            }
+        }
```

Comment:
This should happen after shares are accounted with `shares[marketId] -= redeemedShares`, otherwise this will never remove a market (you can't deallocate an amount > 0 to a market that has no shares)